### PR TITLE
Merge `8.2.x` into `develop`

### PR DESCRIPTION
**Pull request by tleon**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Merge `8.2.x` into `develop`
| Type?             | improvement 
| Category?         | ME 
| BC breaks?        | no
| Deprecations?     |  no
| How to test?      | CI & tests green
| UI Tests          | https://github.com/tleon/ga.tests.ui.pr/actions/runs/10925418491
| Sponsor company   | PrestaShop SA

### CODE REVIEW

File: classes/order/Order.php [hunk 1]
```diff
@@ -746,8 +746,15 @@ protected function setProductCustomizedDatas(&$product, $customized_datas)
      */
     protected function setProductCurrentStock(&$product)
     {
-        $product['current_stock'] = StockAvailable::getQuantityAvailableByProduct($product['product_id'], $product['product_attribute_id'], (int) $this->id_shop);
-        $product['location'] = StockAvailable::getLocation($product['product_id'], $product['product_attribute_id'], (int) $this->id_shop);
+        if (Configuration::get('PS_ADVANCED_STOCK_MANAGEMENT')
```
Comment (Hlavtox) [inline classes/order/Order.php]:
```suggestion
        if (Configuration::get('PS_ADVANCED_STOCK_MANAGEMENT')
```

PS_ADVANCED_STOCK_MANAGEMENT Has been completely removed, you can keep only the else part

Comment (jolelievre) [reply classes/order/Order.php]: Indeed

Comment (Hlavtox) [reply classes/order/Order.php]: @tleon This was the change in the method. ;-)

https://github.com/PrestaShop/PrestaShop/commit/74d5c5c924350feab0e6b3707bdb1e9488252ccf

Just cast location to string.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Just to prevent accidental merge, dismiss when the feedback is implemented

**Review by boherm — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]

File: src/PrestaShopBundle/DependencyInjection/Config/ConfigYamlLoader.php [hunk 2]
```diff
@@ -44,7 +44,8 @@ class ConfigYamlLoader extends FileLoader
     public function load($resource, $type = null)
     {
         $path = $this->locator->locate($resource);
-        $configValues = Yaml::parse(file_get_contents($path));
+//        $configValues = Yaml::parse(file_get_contents($path), Yaml::PARSE_CONSTANT); // new
```
Comment (Hlavtox) [inline src/PrestaShopBundle/DependencyInjection/Config/ConfigYamlLoader.php]: Something weird here?

**Review by Hlavtox — DISMISSED** [review]
See comment

File: classes/Product.php [hunk 3]
```diff
@@ -7304,8 +7308,7 @@ public function getAnchor($id_product_attribute, $with_id = false)
         foreach ($attributes as &$attr) {
             $group = str_replace($sep, $replace, Tools::str2url((string) $attr['group']));
             $name = str_replace($sep, $replace, Tools::str2url((string) $attr['name']));
-            $anchor .= '/' . ($with_id ? (int) $attr['id_attribute'] . $sep : '') . $group . $sep . $name;
-        }
+            $anchor .= '/' . ($with_id && isset($attr['id_attribute']) && $attr['id_attribute'] ? (int) $attr['id_attribute'] . $sep : '') . $group . $sep . $name;        }
```
Comment (kpodemski) [inline classes/Product.php]: Looks like a mistake with the placement of `}`

File: classes/controller/AdminController.php [hunk 4]
```diff
@@ -2067,7 +2067,8 @@ public function initHeader()
                 'employee' => $this->context->employee,
                 'search_type' => Tools::getValue('bo_search_type'),
                 'bo_query' => Tools::safeOutput(Tools::getValue('bo_query')),
-                'quick_access' => empty($quick_access) ? false : $quick_access,
+//                'quick_access' => empty($quick_access) ? [] : $quick_access, // new
+                'quick_access' => empty($quick_access) ? false : $quick_access, // old
```
Comment (kpodemski) [inline classes/controller/AdminController.php]: Mistake?

Comment (tleon) [reply classes/controller/AdminController.php]: There is a lot of fail jobs i'm testing thing to see what make the tests go red.

**Review by boherm — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**